### PR TITLE
platform/miyoo: scripted versioning of IPK info

### PR DESCRIPTION
**Pull request by Apaczer**
well not sure how to aproach that extra btn, so adding one non-existent on these platforms.

### CODE REVIEW

**Comment by irixxxx** [comment]
I'm OK with the packeting change, but what's that MODE button stuff needed for? Do we really need a default binding for the MODE button?

**Comment by Apaczer** [comment]
Wanted to cover whole 3-btn control pad, especially if we have free physical input. There is no MODE btn in def_binds so I'm adding random SDLKey which will not infect other bindings, then swapping it in plat_miyoo - not sure about approach. Regardless a few games use MODE explicitly for special actions beside standard use, so might be worth adding - must test those.

File: platform/opendingux/inputmap.c [hunk 1]
```diff
@@ -137,8 +140,10 @@ void plat_target_setup_input(void)
 	if (strcmp(plat_device, "miyoo") == 0) {
 		/* swapped A/B and X/Y keys */
 		keyswap(SDLK_LALT, SDLK_LCTRL);
+		keyswap(SDLK_MENU, SDLK_ESCAPE);
```
Comment (Apaczer) [inline platform/opendingux/inputmap.c:143]: not working with current setup...so dumping that idea, also missed the point of nameswap() obviously.

**Comment by Apaczer** [comment]
so it's just a 7'th btn on a 6-btn pad...I shall blame official nintendo guys for making 3-btn pad with MODE btn for switch (and my ignorance 😄). Ending here as there is no need for MODE btn for 3-btn input controller. Felt like it would be a shame to leave one key unbinded for miyoo.

**Comment by irixxxx** [comment]
Heh... there are a lot of other devices having additional buttons, think only of L2/R2. I tried to chose a common configuration somehow similar on all supported devices. It's not cast in stone anyway since it's all reconfigurable, so those unbound keys aren't exactly lost.
Maybe one of the PEVB hotkey functions would be better to bind on that key.

**Comment by Apaczer** [comment]
I believe we can leave it as is at least until v2.00 drops, adding other bind from PEVB may be too risky for regular user (aside from vol change).
Anyway I must ask, is there an easier approach for adding new bind for specific plat in inputmap.c without affecting other platforms?

**Comment by irixxxx** [comment]
Only through the setup_input function. There's however no functionality yet to do so, only for remapping. It wouldn't be very difficult though. The exported array sizes should go away for a terminator in the arrays, then there could be some empty template entries at the end to add additional keys if need be.

**Pull request merged**